### PR TITLE
Feature/11 project details

### PR DESCRIPTION
**Pull request by wtang98**
Projects lists and project create form with tests
update on student list sort filter buttons with testing.

### CODE REVIEW

File: src/pages/projects/projects-create/projects-create-top-nav/projects-create-top-nav.jsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+const ProjectCreateTopNav = () => {
+    
+    return (
+        <nav className="top-nav d-flex flex-column justify-content-between">
+            <div className="top-nav__header d-flex align-items-center justify-content-between">
+                <h1 className="top-nav__header-title">Create Project</h1>
+                <div className="top-nav__header-button border-0">
+                    <Link to = "/projects">
+                        <button className="btn-back top-nav__header-button">Go Back</button>
+                    </Link>
```
Comment (mbickell) [inline src/pages/projects/projects-create/projects-create-top-nav/projects-create-top-nav.jsx:14]: Consistent with other TopNav component

File: src/pages/projects/projects-list/__tests__/projects-list.test.js [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+import { BrowserRouter } from "react-router-dom";
+import {within} from '@testing-library/dom';
+
+// Integration tests carried out as it avoids the need to pass down props again. It also simulates user experience more accurately. 
+
+it("should render 8 projects on the page when the page is first rendered", async() => {
+  // 1. Arrange
+  render(
+  <BrowserRouter>
+  <ProjectList/>
```
Comment (mbickell) [inline src/pages/projects/projects-list/__tests__/projects-list.test.js:13]: indentation

File: src/pages/projects/projects-list/projects-list.jsx [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    const filterDataByReact = () => {
+        setLanguageFilter("React");
+        setProjectsData(projectsDataFile.filter(project =>project.language.includes("React")));
+    }
+
+    const filterDataByJava = () => {
+        setLanguageFilter("Java");
+        setProjectsData(projectsDataFile.filter(project =>project.language === "Java"));
+    }
+
```
Comment (mbickell) [inline src/pages/projects/projects-list/projects-list.jsx:41]:
```suggestion
const filterDataByJava = (language) => {
        setLanguageFilter(language);
        setProjectsData(projectsDataFile.filter(project =>project.language === language));
    }
```

```suggestion
filterDataByJava={() => filterDataByJava("Java")}
```

File: src/pages/students/students-list/students-top-nav/students-top-nav.jsx [hunk 4]
```diff
@@ -1,16 +1,16 @@
...
     const {filterDataByCorporate, filterDataByAll, filterDataByFullTime, filterDataBySelfPaced, enrolledFilter} = props;
     return (
         <nav className="top-nav d-flex flex-column justify-content-between">
             <div className="top-nav__header d-flex align-items-center justify-content-between">
                 <h1 className="top-nav__header-title"> Students </h1>
-                {/* <Link to="/studentcreate"> */}
+                <Link to="/student/create">
                     <button className="btn-primary top-nav__header-button border-0 me-2">+ Create</button>
-                {/* </Link> */}
+                </Link>
```
Comment (mbickell) [inline src/pages/students/students-list/students-top-nav/students-top-nav.jsx:13]: Consistency for top navs

**Review by mbickell — CHANGES_REQUESTED** [review]

File: src/pages/projects/projects-list/projects-list.jsx (hunk 3 above)
Comment (angaar96) [reply src/pages/projects/projects-list/projects-list.jsx:41]: done.

**Pull request merged**